addons/binding/org.openhab.binding.tesla/src/main/java/org/openhab/binding/tesla/handler/TeslaHandler.java

```diff
@@ -833,9 +871,33 @@ public void run() {
             if (getThing().getStatus() == ThingStatus.ONLINE) {
                 if (isAwake()) {
                     requestData(TESLA_CHARGE_STATE);
+                    try {
+                        Thread.sleep(1000);
```

Comment:
Better also make a bigger try catch or do you really want to keep running if someone politely asked your thread to interupt?

If we receive this InterruptedException, the interrupted flag was cleared and we most likely need to reset the flag:

http://www.yegor256.com/2015/10/20/interrupted-exception.html
https://www.ibm.com/developerworks/library/j-jtp05236/